xwiki-platform-distribution/xwiki-platform-distribution-flavor/xwiki-platform-distribution-flavor-test/xwiki-platform-distribution-flavor-test-ui/src/test/it/org/xwiki/test/ui/CompareVersionsTest.java

```diff
@@ -276,6 +279,64 @@ public void testUnifiedDiffShowsInlineChanges()
             "+alice = <ins>2 * </ins>bob;", " bob = tmp;");
     }
 
+    /**
+     * Tests navigation between versions.
+     */
+    private void testVersionNavigation()
+    {
+        String queryString = "viewer=changes&rev1=1.2&rev2=6.4";
+        getUtil().gotoPage(getTestClassName(), testPage.getMetaDataValue("page"), "view", queryString);
+        ChangesPane changesPane = new ChangesPane();
+
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version < 1.2 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.4 >"));
+
+        By arrowLeftShiftSelector = By.cssSelector("#changes-info-boxes > a.changes-arrow-left");
+        By arrowRightShiftSelector = By.cssSelector("#changes-info-boxes > a.changes-arrow-right");
+
+        changesPane.clickArrowLeftShift();
+        changesPane = new ChangesPane();
+        assertTrue(changesPane.getFromVersionSummary().startsWith("From version 1.1 >"));
+        assertTrue(changesPane.getToVersionSummary().startsWith("To version < 6.3 >"));
```

Comment:
I'm afraid there has been a misunderstanding on the functionality of the left and right outer "arrows".

The idea is to move to the previous or to the next *change* and not to *shift* the currently defined "changes window". The "changes window" is controlled by the use of the left/right from and left/right to arrows, while the 1-by-1 change navigation is controlled by the outer-most arrows.

In this case:
* the previous change for the window "1.2 > 6.4" should be "1.1 > 1.2"
* the next change for the window "1.2 > 6.4" should be "6.4 > 6.5 (or 7.1, or whatever the next version is, compared to 6.4)"